packages/client/src/clients/in-app-messages/message-client.ts

```diff
@@ -124,7 +130,16 @@ export class InAppMessagesClient {
       (messages, messageId) => {
         const message = state.messages[messageId];
         if (message) {
-          messages.push(message as InAppMessage<TContent, TData>);
+          // Check if the message should be displayed based on engagement timestamps and
+          // the active query filters
+          const isVisible = this.getMessageVisibility(
+            message,
+            this.fetchOptions,
```

Comment:
Right now, the client saves the last used set of filters by updating `this.queryKey` (and `this.fetchOptions`) on initialization and on each fetch, so it should always have the last used filters saved. 

I think we can avoid requiring `queryKey` as a param, since if it's required, that's another thing users have to keep track of and pass back in when getting messages via the selector.

Reply:
Yeah, great questions! Lets discuss how we should keep track of queries per message client. Maybe returning the query key on fetch and initialization then requiring it as a param makes sense so the consumer has to track it (either when implementing manually or in the react-core hooks).

Reply:
As discussed, we'll remove the option to pass additional filters on `fetch()`